Source/WebCore/workers/service/server/SWServerRegistration.cpp

```diff
@@ -423,4 +424,39 @@ std::optional<ExceptionData> SWServerRegistration::setNavigationPreloadHeaderVal
     return { };
 }
 
+void SWServerRegistration::addCookieChangeSubscriptions(Vector<CookieChangeSubscription>&& subscriptions)
+{
+    for (auto& requestedSubscription : subscriptions) {
+        bool isNewSubscription = true;
+        for (auto& currentSubscription : m_cookieChangeSubscriptions) {
+            if (currentSubscription.name == requestedSubscription.name && currentSubscription.url == requestedSubscription.url) {
+                isNewSubscription = false;
+                break;
+            }
+        }
+
+        if (isNewSubscription)
+            m_cookieChangeSubscriptions.append(WTFMove(requestedSubscription));
+    }
+}
+
+void SWServerRegistration::removeCookieChangeSubscriptions(Vector<CookieChangeSubscription>&& subscriptions)
+{
+    for (auto& requestedSubscription : subscriptions) {
+        size_t i = 0;
+        for (i = 0; i < m_cookieChangeSubscriptions.size(); i++) {
+            auto currentSubscription = m_cookieChangeSubscriptions[i];
+            if (currentSubscription.name == requestedSubscription.name && currentSubscription.url == requestedSubscription.url) {
+                m_cookieChangeSubscriptions.remove(i);
+                break;
+            }
+        }
+    }
+}
+
+Vector<CookieChangeSubscription> SWServerRegistration::cookieChangeSubscriptions()
```

Comment:
Should likely:
1. Be inline in the header
2. be const
3. return ` const Vector&`

Reply:
I changed the subscriptions to be stored in a HashSet instead of vector. Since this function now needs to create a vector of the subscriptions and that vector will be destroyed at the end of the function, I have changed the function to be const, but did not inline the function of have it return `const Vector&`. I suppose we could still inline, but the function is no longer just a return statement, so I don't know if that would make much sense. Let me know what you think.